auth-service/src/main/kotlin/uk/gov/communities/delta/auth/utils/PasswordChecker.kt

```diff
@@ -0,0 +1,37 @@
+package uk.gov.communities.delta.auth.utils
+
+import java.util.*
+
+class PasswordChecker {
+
+    private fun containsCommonString(lowerCasePassword: String): Boolean {
+        val commonStrings = arrayOf("qwerty", "abcd", "123456", "password")
+        for (commonString in commonStrings) {
+            if (lowerCasePassword.contains(commonString)) return true
+        }
+        return false
+    }
+
+    private fun hasInsufficientUniqueCharacters(password: String): Boolean {
+        val uniqueCharacters: MutableSet<Char> = HashSet()
+        for (c in password.toCharArray()) uniqueCharacters.add(c)
+        return uniqueCharacters.size < 5
+    }
+
+    private fun containsPartOfUserEmail(lowerCasePassword: String, userEmail: String): Boolean {
+        val wordsInUserEmail =
+            userEmail.lowercase(Locale.getDefault()).split("[\\W0-9_]".toRegex()).dropLastWhile { it.isEmpty() }
+                .toTypedArray()
```

Comment:
```suggestion
            userEmail.lowercase(Locale.getDefault()).split("[\\W0-9_]".toRegex())
```
I think it should work without those